lib/ui/painting/image.cc

```diff
@@ -28,7 +29,10 @@ void CanvasImage::RegisterNatives(tonic::DartLibraryNatives* natives) {
 
 CanvasImage::CanvasImage() {}
 
-CanvasImage::~CanvasImage() {}
+CanvasImage::~CanvasImage() {
+  SkImage* image = image_.release();
+  Threads::IO()->PostTask([image]() { image->unref(); });
```

Comment:
sk_sp should be copyable. So you can probably capture image_ in the block directly.